Gem/Code/Source/ShaderReloadTestComponent.cpp

```diff
@@ -122,22 +126,51 @@ namespace AtomSampleViewer
         }
     }
 
-    bool ShaderReloadSoakTestComponent::ReadInConfig(const AZ::ComponentConfig*)
+    bool ShaderReloadTestComponent::ReadInConfig(const AZ::ComponentConfig*)
     {
         return true;
     }
 
-    void ShaderReloadSoakTestComponent::Activate()
+    void ShaderReloadTestComponent::Activate()
     {
         AZ::TickBus::Handler::BusConnect();
 
         InitTestDataFolders();
+
+        DeleteTestFile("Fullscreen.shader");
+        DeleteTestFile("Fullscreen.azsl");
+
+        AssetStatusTracker assetStatusTracker;
+        assetStatusTracker.StartTracking();
+
+        AZStd::string shaderSourcePath;
+        AzFramework::StringFunc::Path::Join(m_relativeTempSourceFolder.c_str(), "Fullscreen.shader", shaderSourcePath);
+        assetStatusTracker.ExpectAsset(shaderSourcePath, 1);
+
         CopyTestFile(RedShaderFile, "Fullscreen.azsl");
         CopyTestFile("Fullscreen.shader.txt", "Fullscreen.shader");
         m_expectedPixelColor = RED_COLOR;
 
         // A short wait is necessary as the pipeline creation will fatally fail
         // if the "Fullscreen.azshader" doesn't exist.
+        const uint32_t MaxWaitTimeMillis = 10000;
+        const uint32_t MilliWaits = 50;
+        uint32_t timeWaitMillis = 0;
+        while (!assetStatusTracker.DidExpectedAssetsFinish())
+        {
+            AZStd::this_thread::sleep_for(AZStd::chrono::milliseconds(MilliWaits));
+            timeWaitMillis += MilliWaits;
+            if (timeWaitMillis >= MaxWaitTimeMillis)
+            {
+                AZ_Error(LogName, false, "Failed to activate this test because exceeded wait time of %u milliseconds", MaxWaitTimeMillis);
+                return;
+            }
+        }
+        assetStatusTracker.StopTracking();
```

Comment:
It's a bummer to block the whole UI while waiting for the shader to compile. Can this be done in an async way? You could make the AssetStatusTracker be a member and check DidExpectedAssetsFinish() in OnTick instead.

Reply:
Or factor out MaterialHotReloadTestComponent::GetTestAssetStatus

Reply:
The clean solution is to use AsyncAssetLoader.